File: test-plugins/org.yakindu.sct.simulation.core.sexec.test/test-gen/org/yakindu/sct/simulation/core/sexec/test/CastExpressionsTest.java
/**
 * Copyright (c) 2015 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.simulation.core.sexec.test;
import org.eclipse.xtext.junit4.InjectWith;
import org.eclipse.xtext.junit4.XtextRunner;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.yakindu.sct.model.sexec.ExecutionFlow;
import org.yakindu.sct.model.sexec.interpreter.test.util.AbstractExecutionFlowTest;
import org.yakindu.sct.model.sexec.interpreter.test.util.SExecInjectionProvider;
import org.yakindu.sct.test.models.SCTUnitTestModels;
import com.google.inject.Inject;
import static org.junit.Assert.assertTrue;
/**
 *  Unit TestCase for CastExpressions
 */
@SuppressWarnings("all")
@RunWith(XtextRunner.class)
@InjectWith(SExecInjectionProvider.class)
public class CastExpressionsTest extends AbstractExecutionFlowTest {
	@Before
	public void setup() throws Exception {
		ExecutionFlow flow = models
				.loadExecutionFlowFromResource("CastExpressions.sct");
		initInterpreter(flow);
	}
	@Test
	public void CastExpressionTest() throws Exception {
		interpreter.enter();
		assertTrue(getReal("realValue") == 5l);
		assertTrue(getInteger("intValue") == 5l);
		interpreter.runCycle();
		assertTrue(getReal("realValue") == 15l);
		interpreter.runCycle();
		assertTrue(isStateActive("C"));
		assertTrue(getReal("realValue") == 757l);
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/CastExpressionsTest/CastExpressionsTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "CastExpressions.h"

CastExpressions handle;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		castExpressions_init(&handle);
	}
};


TEST_F(StatemachineTest, CastExpressionTest) {					
	castExpressions_enter(&handle);
	EXPECT_TRUE(castExpressionsIface_get_realValue(&handle)== 5l);
	EXPECT_TRUE(castExpressionsIface_get_intValue(&handle)== 5l);
	castExpressions_runCycle(&handle);
	EXPECT_TRUE(castExpressionsIface_get_realValue(&handle)== 15l);
	castExpressions_runCycle(&handle);
	EXPECT_TRUE(castExpressions_isStateActive(&handle, CastExpressions_main_region_C));
	EXPECT_TRUE(castExpressionsIface_get_realValue(&handle)== 757l);
}


